queuelistener/listener.go

```diff
@@ -196,6 +201,15 @@ func Listen(o Options) (net.Listener, error) {
 		return nil, err
 	}
 
+	if o.ProxyProtocol {
+		log.Printf("Proxy protocol listener")
+		pl := &proxyproto.Listener{
+			Listener:          nl,
+			ReadHeaderTimeout: 10 * time.Second,
+		}
+		return listenWith(pl, o)
+	}
```

Comment:
I'm not sure how it will be taken by original team of `zalando/skipper`. But IMO basically you don't need to allocate new variable. You can allocate your `proxyproto.Listener` to existing variable and pass current `net.Listener` as an internal `Listener` to the `proxyproto.Listener`.
```suggestion
		nl = &proxyproto.Listener{
			Listener:          nl,
			ReadHeaderTimeout: 10 * time.Second,
		}
	}
```